deps/tools/common.mk

```diff
@@ -167,19 +174,29 @@ $$(build_staging)/$2.tar: $$(BUILDDIR)/$2/build-compiled
 	cd $$(build_staging)/$2$$(build_prefix) && $$(TAR) -cf $$@.tmp .
 	rm -rf $$(build_staging)/$2
 	mv $$@.tmp $$@
+$$(build_staging)/$2.tar.xz: $$(build_staging)/$2.tar
+	$$(XZ) -z -T0 $$<
 
 UNINSTALL_$(strip $1) := $2 staged-uninstaller
 
+ifeq ($(STAGING_EXTENSION),tar)
```

Comment:
This is a bit unfortunate, but we can't have a rule that produces a .tar from a .tar.xz because that would conflict with the .tar.xz <- .tar rule above (introducing a circular definition). Maybe somebody with some more GNU Make fu knows how to do this better?